src/lib/Leaf.test.tsx

```diff
@@ -597,4 +563,59 @@ test("get errors for root location via undefined location", async () => {
     await waitFor(() => {
         expect(queryAllByTestId("top-level-error").map(node => node.textContent)).toEqual([]);
     });
+});
+
+test("does not re-render when leaf value and state has not changed", () => {
+    let renders: Array<string> = [];
+
+    const Wrapper = () => {
+        const [model, setModel] = useState({
+            person: {
+                firstName: "Stewart",
+                lastName: "Anderson"
+            }
+        });
+
+        const firstName = React.useMemo(() => (firstName: any, updateFirstName: (updatedModel: any) => void): JSX.Element => {
+            renders.push("firstName");
+            return <TextInput
+                data-testid="firstName"
+                value={firstName}
+                onChange={updateFirstName} />;
+        }, []);
+        const lastName = React.useMemo(() => (lastName: any, updateLastName: (updatedModel: any) => void): JSX.Element => {
+            renders.push("lastName");
+            return <TextInput
+                data-testid="lastName"
+                value={lastName}
+                onChange={updateLastName} />;
+        }, []);
+
+        return <>
+            <Leaf location="person.firstName" model={model} onChange={setModel}>
+                {firstName}
+            </Leaf>
+            <Leaf location="person.lastName" model={model} onChange={setModel}>
+                {lastName}
+            </Leaf>
+        </>;
```

Comment:
This might be worth adding to the readme.

Child render function will be called on each render because you could have a child Leaf (or something else) that may need to re-render even if the parent Leaf does not.